DNSServiceDiscoveryUnicast/README.md

```diff
@@ -0,0 +1,136 @@
+# Unicast DNS service discovery
+
+This application demonstrates how to perform [DNS service discovery](https://learn.microsoft.com/azure-sphere/app-development/service-discovery) by sending DNS-SD queries to a configured DNS server. It is based on the [Multicast DNS service discovery sample](https://github.com/Azure/azure-sphere-samples/tree/main/Samples/DNSServiceDiscovery).
+
+The application queries a configured DNS server for **PTR** records that identify instances of the `_http._tcp` service. If available, the application then queries the server for the **SRV**, **TXT**, and **A** records that contain the DNS details for the service instance. Once complete, the Azure Sphere firewall allows the application to connect to the discovered host names, and a simple curl HTTP fetch is performed to the hostname returned by the **SRV** record.
+
+A container is provided for a simple, minimal DNS server, configured to serve the `home` domain, with an `_http._tcp` service pointing at `HelloWorld._http._tcp.home`, which in turn is configured to resolve to `www.dns-sd.org`.
+
+Unlike multicast service discovery, the discovered service may exist outside of the local network for the device.
+
+## Contents
+
+| File/folder | Description |
+|-------------|-------------|
+| `.vscode/` | Settings for VSCode |
+| `container/` | Dockerfile and supporting content for DNS server container |
+| `main.c`, `dns-sd.c`, `dns-sd.h`       | source code. |
+| `eventloop_timer_utilities.c`, `eventloop_timer_utilities.h` | common library source code |
+| `app_manifest.json` | The application manifest |
+| `CMakeLists.txt` | Contains the project information and produces the build. |
+| `CMakeSettings.json` | Configures CMake with the correct command-line options. |
+| `README.md` | This README file. |
+| `LICENSE.txt`   | The license for the project. |
+
+## Prerequisites
+
+The project requires the following hardware:
+
+- An [Azure Sphere development board](https://aka.ms/azurespheredevkits)
+- (optional) a machine running Docker to run the DNS server container
+
+## Setup
+
+1. Even if you've performed this set up previously, ensure that you have Azure Sphere SDK version 22.11 or above. At the command prompt, run **azsphere show-version** to check. Install [the Azure Sphere SDK](https://learn.microsoft.com/azure-sphere/install/install-sdk) as needed.
+1. Connect your Azure Sphere device to your computer by USB.
+1. Connect your Azure Sphere device to the same local network as the DNS service.
+1. Enable application development, if you have not already done so, by entering the **azsphere device enable-development** command at the command prompt.
+
+1. Set up a DNS service. A container is provided that runs a minimal `bind9` on Ubuntu 22.04:
```

Comment:
There's an assumption that the user knows how to discover the IP address of their docker container, is there something we can point them to?

Reply:
The `docker run` command binds the container's DNS ports to the host's, so the IP address of the docker container doesn't matter - it's the IP address of the host running the container that will appear as the DNS server. I'll clarify this.